tests/components/seventeentrack/test_sensor.py

```diff
@@ -0,0 +1,327 @@
+        CONF_PASSWORD: 'test'
+    }
+}
+
+INVALID_CONFIG = {
+    'sensor': {
+        'platform': 'seventeentrack',
+        'boom': 'test',
+    }
+}
+
+VALID_CONFIG_FULL = {
+    'sensor': {
+        'platform': 'seventeentrack',
+        CONF_USERNAME: 'test',
+        CONF_PASSWORD: 'test',
+        CONF_SHOW_ARCHIVED: True,
+        CONF_SHOW_DELIVERED: True
+    }
+}
+
+VALID_CONFIG_FULL_NO_DELIVERED = {
+    'sensor': {
+        'platform': 'seventeentrack',
+        CONF_USERNAME: 'test',
+        CONF_PASSWORD: 'test',
+        CONF_SHOW_ARCHIVED: False,
+        CONF_SHOW_DELIVERED: False
+    }
+}
+
+DEFAULT_SUMMARY = {
+    "Not Found": 0,
+    "In Transit": 0,
+    "Expired": 0,
+    "Ready to be Picked Up": 0,
+    "Undelivered": 0,
+    "Delivered": 0,
+    "Returned": 0
+}
+
+NEW_SUMMARY_DATA = {
+    "Not Found": 1,
+    "In Transit": 1,
+    "Expired": 1,
+    "Ready to be Picked Up": 1,
+    "Undelivered": 1,
+    "Delivered": 1,
+    "Returned": 1
+}
+
+
+class ClientMock:
+    """Mock the py17track client to inject the ProfileMock."""
+
+    def __init__(self, websession) -> None:
+        """Mock the profile."""
+        self.profile = ProfileMock()
+
+
+class ProfileMock:
+    """ProfileMock will mock data coming from 17track."""
+
+    package_list = []
+    login_result = True
+    summary_data = DEFAULT_SUMMARY
+    account_id = '123'
+
+    @classmethod
+    def reset(cls):
+        """Reset data to defaults."""
+        cls.package_list = []
+        cls.login_result = True
+        cls.summary_data = DEFAULT_SUMMARY
+        cls.account_id = '123'
+
+    def __init__(self) -> None:
+        """Override Account id."""
+        self.account_id = self.__class__.account_id
+
+    async def login(self, email: str, password: str) -> bool:
+        """Login mock."""
+        return self.__class__.login_result
+
+    async def packages(self, package_state: Union[int, str] = '',
+                       show_archived: bool = False) -> list:
+        """Packages mock."""
+        return self.__class__.package_list[:]
+
+    async def summary(self, show_archived: bool = False) -> dict:
+        """Summary mock."""
+        return self.__class__.summary_data
+
+
+@pytest.fixture(autouse=True, name="mock_py17track")
+def fixture_mock_py17track():
+    """Mock py17track dependency."""
+    with MockDependency('py17track'):
+        yield
+
+
+@pytest.fixture(autouse=True, name="mock_client")
+def fixture_mock_client(mock_py17track):
+    """Mock py17track client."""
+    with mock.patch('py17track.Client', new=ClientMock):
+        yield
+    ProfileMock.reset()
+
+
+@pytest.fixture(name="hass")
+def fixture_hass(mock_client):
+    """Create hass instance and clean it."""
+    hass = get_test_home_assistant()
+    hass.components.persistent_notification = mock.MagicMock()
+    yield hass
+    hass.stop()
+    storage_dir = get_test_config_dir('.storage')
+    if os.path.isdir(storage_dir):
+        shutil.rmtree(storage_dir)
```

Comment:
Mock the storage instead if needed with `mock_storage` in tests/common.py.

Reply:
Yep, already tried `mock_storage` and/or `mock_registry` but I always get:

```
***** ERROR
Tests are leaving files behind. Please update the tests to avoid writing any files:
tests/testing_config/.storage/core.entity_registry
```

here is the content after running all tests:

```json
{
    "data": {
        "entities": [
            {
                "config_entry_id": null,
                "device_id": null,
                "disabled_by": null,
                "entity_id": "sensor.seventeentrack_packages_delivered",
                "name": null,
                "platform": "seventeentrack",
                "unique_id": "summary_123_delivered"
            },
            {
                "config_entry_id": null,
                "device_id": null,
                "disabled_by": null,
                "entity_id": "sensor.seventeentrack_packages_returned",
                "name": null,
                "platform": "seventeentrack",
                "unique_id": "summary_123_returned"
            },
            {
                "config_entry_id": null,
                "device_id": null,
                "disabled_by": null,
                "entity_id": "sensor.seventeentrack_packages_in_transit",
                "name": null,
                "platform": "seventeentrack",
                "unique_id": "summary_123_in_transit"
            },
            {
                "config_entry_id": null,
                "device_id": null,
                "disabled_by": null,
                "entity_id": "sensor.seventeentrack_packages_ready_to_be_picked_up",
                "name": null,
                "platform": "seventeentrack",
                "unique_id": "summary_123_ready_to_be_picked_up"
            },
            {
                "config_entry_id": null,
                "device_id": null,
                "disabled_by": null,
                "entity_id": "sensor.seventeentrack_packages_expired",
                "name": null,
                "platform": "seventeentrack",
                "unique_id": "summary_123_expired"
            },
            {
                "config_entry_id": null,
                "device_id": null,
                "disabled_by": null,
                "entity_id": "sensor.seventeentrack_packages_undelivered",
                "name": null,
                "platform": "seventeentrack",
                "unique_id": "summary_123_undelivered"
            },
            {
                "config_entry_id": null,
                "device_id": null,
                "disabled_by": null,
                "entity_id": "sensor.seventeentrack_packages_not_found",
                "name": null,
                "platform": "seventeentrack",
                "unique_id": "summary_123_not_found"
            }
        ]
    },
    "key": "core.entity_registry",
    "version": 1
}
```
any idea ?

Reply:
How did you apply `mock_storage`?

Reply:
Had it to work with:

 ```python
@pytest.fixture(name="hass")
def fixture_hass(mock_client, hass_storage):
    """Create hass instance and clean it."""
    hass = get_test_home_assistant()
    hass.components.persistent_notification = mock.MagicMock()
    yield hass
    hass.stop()
```